azure/function-app/function_app.tf

```diff
@@ -45,16 +44,47 @@ resource "azurerm_windows_function_app" "function_app" {
   dynamic "identity" {
     for_each = var.identity != null ? [var.identity] : []
     content {
-      type         = "UserAssigned"
+      type         = var.assign_system_identity ? "SystemAssigned, UserAssigned" : "UserAssigned"
       identity_ids = [data.azurerm_user_assigned_identity.identity.0.id]
     }
   }
 
+  dynamic "identity" {
+    for_each = var.assign_system_identity && var.identity == null ? [var.assign_system_identity] : []
+    content {
+      type = "SystemAssigned"
+    }
+  }
+
   site_config {
     application_insights_key = var.insights.enabled ? azurerm_application_insights.app_insights.0.instrumentation_key : null
 
-    application_stack {
-      node_version = "~18"
+    dynamic "application_stack" {
+      for_each = var.runtime.name == "dotnet" ? [var.runtime] : []
+      content {
+        dotnet_version = application_stack.value.version
+      }
+    }
+
+    dynamic "application_stack" {
+      for_each = var.runtime.name == "java" ? [var.runtime] : []
+      content {
+        java_version = application_stack.value.version
+      }
+    }
+
+    dynamic "application_stack" {
+      for_each = var.runtime.name == "node" ? [var.runtime] : []
+      content {
+        node_version = application_stack.value.version
+      }
+    }
+
+    dynamic "application_stack" {
```

Comment:
python is missing if I see it correctly?

see [here](https://learn.microsoft.com/en-us/azure/azure-functions/supported-languages?culture=en-us&tabs=isolated-process%2Cv4&pivots=programming-language-python)

Reply:
We only use a `azurerm_windows_function_app` at the moment and unfortunately python is not supported on windows 😭 

implementing `azurerm_linux_function_app` is out of the scope of this issue, created a [new one](https://github.com/THEY-Consulting/they-terraform/issues/52) for this